extensions/rich_text_components/protractor.js

```diff
@@ -17,6 +17,10 @@
  * protractor tests.
  */
 
+// NOTE to devs: If other functions or constants are added above this constant,
```

Comment:
Seems way too fragile. Why not have the backend test search for "var RICH_TEXT_COMPONENTS = {\n" instead? Then look for the next '}' after that, and split by ',\n' to get the individual lines. I think that will allow you to simplify a fair bit of the parsing logic, and then people who edit this file won't need to be super-careful about where to put things..

Reply:
Done, made the initial search to be for 'varRICH_TEXT_COMPONENTS={' since \n and ' ' are removed  before.